tests/hazmat/backends/test_openssl.py

```diff
@@ -453,6 +454,28 @@ class FakeAlgorithm(object):
             backend.create_cmac_ctx(FakeAlgorithm())
 
 
+class TestOpenSSLCreateX509CSR(object):
+    @pytest.mark.skipif(
+        backend._lib.OPENSSL_VERSION_NUMBER >= 0x10001000,
+        reason="Requires an older OpenSSL. Must be < 1.0.1"
+    )
+    def test_unsupported_dsa_keys(self):
+        private_key = DSA_KEY_2048.private_key(backend)
+
+        with pytest.raises(NotImplementedError):
+            backend.create_x509_csr(object(), private_key, hashes.SHA1())
+
+    @pytest.mark.skipif(
+        backend._lib.OPENSSL_VERSION_NUMBER >= 0x10001000,
+        reason="Requires an older OpenSSL. Must be < 1.0.1"
+    )
+    def test_unsupported_ec_keys(self):
+        private_key = ec.generate_private_key(ec.SECT283K1(), backend)
```

Comment:
Also, this should probably be secp256r1 as it's a more widely supported curve across the openssl versions we support.